src/traits.rs

```diff
@@ -3,6 +3,14 @@ pub use crate::data::DataOrSuffStat;
 pub use crate::suffstat_traits::*;
 use rand::Rng;
 
+pub trait Parameterized {
+    type Parameters;
+
+    fn emit_params(&self) -> Self::Parameters;
+
+    fn from_params(params: Self::Parameters) -> Self;
```

Comment:
Since the implementations are proper `struct`s, would it be better to drop these and just use the standard `From` trait each way?